core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/HDFSAction.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.batch.action;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.action.Action;
+import co.cask.cdap.etl.api.action.ActionContext;
+import org.apache.commons.io.filefilter.WildcardFileFilter;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileFilter;
+import java.io.IOException;
+import javax.annotation.Nullable;
+
+/**
+ * Move file(s) within HDFS.
+ * A user must specify file/directory path and destination file/directory path
+ */
+public class HDFSAction extends Action {
+  private HDFSActionConfig config;
+
+  private static final Logger LOG = LoggerFactory.getLogger(HDFSAction.class);
+
+  public HDFSAction(HDFSActionConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void run(ActionContext context) throws Exception {
+    //determine how many times we have to run this
+    File folder = new File(config.sourcePath);
```

Comment:
Can the `config.sourcePath` point to a single file, instead of a folder? Variable name seems to imply its always a directory. Also, what code changes would be required to support a single file as the source (i.e. we can't call `listFiles` below).